### ISSUE TITLE
How to help people who have questions about EPUBs?

### ISSUE THREAD

**Issue by dauwhe** [w3c/publ-cg#88]
EPUBs are complicated. The spec doesn't fit on an index card. There are more reading systems then Democratic candidates for President. Book content is infinitely varied.

So where do people get help? The IDPF Forums seem [mostly-dead](http://idpf.org/forums). Stack Overflow doesn't really seem to cover EPUB. Best Practices seem devoted to answering questions before we know if the question is being asked. 

Is there anything this CG can do about that? In some other communities, there is a forum where people ask questions, and other people answer them, or at least try to help. What would happen if we started a repo where you could ask any question about EPUB?

**Comment by artbyrt** [comment]
Great idea! +1


Ruth Tait
http://artbyrt.com

**Comment by damiangza** [comment]
Yes, +1. Would be great to have one location where the latest and up to date solutions can be found. Googling is only useful upto a point.

**Comment by rdeltour** [comment]
Thanks @dauwhe for launching the discussion. This is an issue with EPUBCheck too, where we regularly get support requests or questions asked on our issue tracker, which is not made for that.

> Stack Overflow doesn't really seem to cover EPUB.

I guess this could change but we need to make it happen, as a community. Users need something that is both visible and easy to access (e.g. no complex login or frightening user interface). I'm wondering if trying to get an active community on StackOverflow would be more suited than a GitHub repo?

**Comment by llemeurfr** [comment]
I tend to agree that Stack Overflow, even if many people don't like it, is a well known platform with an excellent SEO, and it would be foolish to crate a Q/A platform of our own. And Github is IMHO not a proper tool for EPUB users.

It is not sufficient, as the community also needs long format articles. isn't "EPUB Secrets" is the proper platform to promote? Who is handling this platform is fact?

Laurent

**Comment by mattgarrish** [comment]
If there are lessons to be learned from the idpf forums, it's that getting a critical mass of people answering questions is no easy feat, and that making people aware that such forums even exist is equally problematic.

It might be helpful to look into the various platforms where people are asking epub questions and see whether there is a clear favourite and/or what could be done to improve the experience/responsiveness. 

Is there a way to leverage discussions that already happen in Twitter, for example? (i.e., to port answers somewhere where they can actually be found).

Is Stack Overflow good enough if we just get more people involved in answering questions?

**Comment by teytag** [comment]
Hi,

This subject is a important. I used IDPF Forums for a long time. I found it quite useful. But login with membership has created a problem for a new question and answer. 

Stack Overflow is quite useful for any developers. But this is a website that I did not get help with until this time EPUB. Because generally there are errors in the coding provided for the answers. In short, it doesn't give me much confidence. However, IDPF Forums is an enterprise platform where experts respond.

I would like to add another subject. Can basic pages about EPUB3 be opened at https://www.w3schools.com/?

As a result, can an improvement be made for IDPF Forums without membership mandatory? http://idpf.org/forums is a source address.

Best regards

N. Erhan Uzumcu
Art Director and EPUB Designer
TEYTAG Founder
(Turkish Electronic Publishing Design and Research Group - Istanbul, Turkey)
https://independent.academia.edu/NErhanÜzümcü
http://dbookmarksblog.blogspot.com.tr
http://teytagblog.blogspot.com.tr
@ePub3Lib #teytag

**Comment by TzviyaSiegman** [comment]
There is an effort underway to document EPUB on [MDN](https://developer.mozilla.org/en-US/), but that doesn't solve the problem of answering questions as they arise. I kind of like the idea of keeping questions and answers on StackOverflow or GitHub. We just need to make sure that we (exactly who TBD) are paying attention.

**Comment by thinkulum** [comment]
A couple of Q&A sites I follow:

- There's a Stack Exchange site for ebooks that has an [EPUB tag](https://ebooks.stackexchange.com/questions/tagged/epub).
- The MobileRead forums have an [EPUB forum](https://www.mobileread.com/forums/forumdisplay.php?f=179).

The activity in these is fairly low, but they have some serious participants with technical knowledge, so there are already people who think of them as places to go for authoring help. One or both of those could be good places to invest in.

**Comment by RachelComerford** [comment]
I would argue that StackOverflow _might not_ be the right place for it. 

While the people who could contribute the answers to the questions that would come up would know StackOverflow as a resource they're not the target audience of this solution. The people we're trying to solve for have little to no development or EPUB experience. They, like me, don't know what StackOverflow is and would never look to it as a resource. They are also unlikely to look for MDN which I say sadly as part of the team working on the documentation. SEO will be key to this no matter where we put resources but I think it's important that we keep perspective on Dave's question. We need to help all users not the already experienced or semi-experienced developers. And as we push a broader audience into this format, that means a broad base of wildly inexperienced beginners with a lot of questions who have never worked with developer resources - they have always worked in word and/or adobe resources.

**Comment by amandasramalho** [comment]
Hi all!
Here in Brazil we also do not have a single place to answer questions. Just like in science, we end up creating "invisible colleges" where we share, among professionals, some news and discoveries about EPUB. As many university publishers started producing EPUB files with weaknesses, we made a complete guide to the format explaining the use of each of the tags (more than 70 pages long). Another "tool" is to use the #eprdctn hashtag of Twitter.
One comment I would like to make: think of a multilingual platform, so that, whatever the language, discussions can be found in one place.

**Comment by dauwhe** [comment]
> I would argue that StackOverflow _might not_ be the right place for it.
> 
> While the people who could contribute the answers to the questions that would come up would know StackOverflow as a resource they're not the target audience of this solution. The people we're trying to solve for have little to no development or EPUB experience. They, like me, don't know what StackOverflow is and would never look to it as a resource. They are also unlikely to look for MDN which I say sadly as part of the team working on the documentation. SEO will be key to this no matter where we put resources but I think it's important that we keep perspective on Dave's question. We need to help all users not the already experienced or semi-experienced developers. And as we push a broader audience into this format, that means a broad base of wildly inexperienced beginners with a lot of questions who have never worked with developer resources - they have always worked in word and/or adobe resources.

One big problem is infrastructure. I feel like we don't have a lot of choices. We can use GitHub issues in this or another repository. We can take advantage of existing sites like StackOverflow. But I don't think we can build our own forum. 

Any existing or new system will be challenging for newcomers. They don't know what's out there, so we have to help them find it, and then help them use it. Having good tutorials will be really important, and even some way of getting people help for logging in to whatever system we go with.

**Comment by rdeltour** [comment]
> While the people who could contribute the answers to the questions that would come up would know StackOverflow as a resource they're not the target audience of this solution.

It's true that StackOverflow’s primary audience is software developers, and the existing topics (that one would see on the home page on their first visit) can be daunting for our target audience.

If we want to dig deeper in the “StackOverflow universe”, using a specialized space like the [ebooks StackExchange community](https://ebooks.stackexchange.com) mentioned by @thinkulum might be better. SEO might not be as good as the main StackOverflow site, but it could be a good comprise. To be investigated…

**Comment by rkwright** [comment]
As a long-time forum poster and monitor, I tend to agree with Romain.  SO is not friendly. Though https://ebooks.stackexchange.com might be a little obscure, at least initially, I like it as a direction.

**Comment by RachelComerford** [comment]
As someone with literally no knowledge in this area, is there anything we can do for SEO of https://ebooks.stackexchange.com?

**Comment by MyDK** [comment]
My experience is this: 
Everybody knew about the IDPFs site, unfortunately it died.
Ebook authors are creating EPUB files using different software/tools and tend to look for help in those environments /forums:
- Ebook authors creating EPUBs with InDesign will look for help in Adobe InDesigns help forum and epubsecrets (last is primarily for Mac users). 
- Ebook authors creating EPUBs using Sigil will look for help on the MobileRead forum. (which works nicely)
- flightdeck have a nice litte handbook: https://ebookflightdeck.com/handbook with code snippets (by joshua tallent.

Having one joint (multi lingual) community site for Best Practice, would be lovely to have. And I would love to contribute to this, answering and helping other ebook authors.

**Comment by artbyrt** [comment]
Here is a suggested solution from a colleague:

mattermost
https://mattermost.com/mattermost-vs-slack/

Ruth Tait
http://artbyrt.com

**Comment by RachelComerford** [comment]
In the interest of moving this forward, a proposal:

Community Group Members begin to formally use https://ebooks.stackexchange.com/ as a Q&A space for EPUB help and to publicly push it as a forum for others to use as well. The best practices sub-group will link to https://ebooks.stackexchange.com/ from the MDN documentation.

Please thumbs up for agreement
Please thumbs down for disagreement

**Comment by artbyrt** [comment]
+1

Ruth Tait

**Comment by damiangza** [comment]
+1
Damian Gibbs

**Comment by llemeurfr** [comment]
+1 to start using  https://ebooks.stackexchange.com/ <https://ebooks.stackexchange.com/> as a Q&A space for EPUB help and to publicly push it as a forum for others to use as well. 
	Rationale: dedicated to ebooks, free, Q&A in Creative Commons, interesting SEO. 

The second part is unclear:
> The best practices sub-group will link to https://ebooks.stackexchange.com/ <https://ebooks.stackexchange.com/> from the MDN documentation.
> 
Before asserting MDN will be the community location for best practices, we should have an overview of the first results of efforts made on MDN by community members, incl.  Rachel Comerford. 

Cordialement, 

Laurent Le Meur
CTO EDRLab

**Comment by mattgarrish** [comment]
I'm hesitant to say any solution is the right one without hearing more from people about what they're looking for. The stackexchange site looks fine if it's just about posing questions and up-voting answers, but it's not designed for discussing problems or solutions.

**Comment by MyDK** [comment]
How about dividing people into 2 help sites/areas, depending on what kind of EPUB help they are looking for: 
1) ordinary readers, users of ebooks and reading systems, and
2) ebook authors / developers who are producing the epub files

**Comment by raducoravu** [comment]
+1 somebody asked for my help to create an EPUB containing an HTML document with a Javascript inside it, it would have been nice to be able to direct them to some kind of users list where people who work more with the EPUB specs can help with such cases.
